### DISCUSSION TITLE
Pretrained model of MSA transformer in the Supervised Contact Prediction

### DISCUSSION THREAD

**Discussion by quailwwk** [facebookresearch/esm#60] [Q&A] [answered]
Hey guy, I'm deeply inspired by your excellent work (MSA transformer), expecially the tied attn.
But it's unfortunately that I can't reproduce the supervised results mentioned in $4.2, based on the released pretrained model of MSA transformer. 
So I wonder if the params in transformer was also updated during supervised training? Thanks!

**Comment by tomsercu** [comment, marked as answer]
Hi quailwwk, thanks that is great to hear!
From the paper, Sec 4.2:
> We evaluate the MSA Transformer as a component of a
> supervised structure prediction pipeline. Following Rives
> et al. (2020), we train a deep residual network with 32 preactivation blocks, each with a filter size of 64, using learning
> rate 0.001. The network is supervised with binned pairwise
> distance distributions (distograms) using the trRosetta training set (Yang et al., 2019) of 15,051 MSAs and structures 

Please let us know if you have any specific issues, and we can assist here.

**Reply by nasserhashemi** [reply]
Hi Tom; 
I have a question regarding this issue:
So you used the same architecture and hyperparameters such as tRosetta instead of number of layers ( since they used 61 and you used 32); and after training the distance map prediction, only use binary classification to test the target to get contact map. is that correct? Also, could you please let me know  Why you did not show distance map ( binned ragne of distance) results such as what trosetta or google-alphafold-1 has done?

**Reply by quailwwk** [reply]
Hi Tom, thanks to your reply.
I have read the supervise section in the paper but I'm still confused what role the MSA transformer plays in the training process of supervised distance network. What does “We evaluate the MSA Transformer as a component of a supervised structure prediction pipeline” mean?  I have two different understandings of this discription:

1) You just use the features extracted from pretrained MSA transformer as the input features of the ResNet network. The supervised training only updates params in ResNet.
2) You connect MSA transformer with ResNet and train them together.

I tried as 1) based on the pretrained MSA transformer but see no improvement.  And there are other small questions: 

- How is “the output embedding of the query sequence” obtained? By feeding the entire MSA into MSA transformer and extract the 1st row of output? Or just feeding the sequence alone into MSA transformer?
- Does "the symmetrized row self-attention maps" contain maps from all heads in all 12 layers? Or only the last layer?

Thanks!

**Reply by tomsercu** [reply]
Yes we use (1) fixed MSA Transformer, not finetuned through. Using trRosetta dataset with MSAs. Some ongoing investigation about dropout on/off during forward pass of the MSA Transformer (it was originally trained with dropout on, but cheaper to dump features once).

> feeding the entire MSA into MSA transformer and extract the 1st row of output?

correct 

> symmetrized row self-attention maps" contain maps from all heads in all 12 layers

correct
